File: fern/snippets/tool-calling/versioning/python/sdk-initialization.py
from composio import Composio

# Pin specific versions for each toolkit
composio = Composio(
    api_key="YOUR_API_KEY",
    toolkit_versions={
        "github": "20250116_00",
        "slack": "20250115_01",
        "gmail": "20250115_00"
    }
)

File: fern/snippets/tool-calling/versioning/typescript/sdk-initialization.ts
import { Composio } from "@composio/core";

// Pin specific versions for each toolkit
const composio = new Composio({
    apiKey: "YOUR_API_KEY",
    toolkitVersions: {
        github: "20250116_00",
        slack: "20250115_01",
        gmail: "20250115_00"
    }
});